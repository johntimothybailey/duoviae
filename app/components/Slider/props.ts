import { SliderProps } from '@react-native-community/slider'

interface Props extends SliderProps {

}

export default Props
