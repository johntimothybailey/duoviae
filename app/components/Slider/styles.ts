import { Spacing } from '../../theme'

const SliderDefault = {
  width: 190
}

const Value = {
  minWidth: 20,
  paddingLeft: Spacing.TINY
}

export default {
  SliderDefault,
  Value
}
