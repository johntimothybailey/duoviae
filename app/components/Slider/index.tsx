import React from 'react'
import Props from './props'
import Slider from '@react-native-community/slider'
import { Text, useTheme } from '@ui-kitten/components'
import styles from './styles'

export default (props: Props): React.ReactElement => {
  const theme = useTheme()
  let style = styles.SliderDefault
  if (props.style) {
    // @ts-expect-error
    style = { ...style, ...props.style }
  }
  return (
    <>
      {/* @ts-expect-error */}
      <Slider
        style={style}
        minimumTrackTintColor={theme['color-primary-500']}
        {...props}
      />
      <Text style={styles.Value}>{props.value}</Text>
    </>
  )
}
