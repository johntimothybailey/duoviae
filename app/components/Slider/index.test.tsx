import React from 'react'
import { cleanup, render } from '@testing-library/react-native'
import { BootComponent } from '../../../test'
import Slider from './index'

describe('Slider', () => {
  describe('Render', () => {
    afterEach(cleanup)
    it('value is displayed', () => {
      const { queryByText } = render(
        <BootComponent>
          <Slider value={6} />
        </BootComponent>
      )
      expect(queryByText('6')).toBeTruthy()
    })
    it('style is set', () => {
      const { toJSON } = render(
        <BootComponent>
          <Slider style={{ width: 300 }} />
        </BootComponent>
      )
      expect(toJSON()).toMatchSnapshot()
    })
  })
})
