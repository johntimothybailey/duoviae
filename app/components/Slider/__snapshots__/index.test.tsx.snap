// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Render style is set 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNCSlider
      disabled={false}
      enabled={true}
      inverted={false}
      maximumValue={1}
      minimumTrackTintColor="#3366FF"
      minimumValue={0}
      onChange={null}
      onRNCSliderSlidingComplete={null}
      onRNCSliderSlidingStart={null}
      onRNCSliderValueChange={null}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      step={0}
      style={
        Array [
          Object {
            "height": 40,
          },
          Object {
            "width": 300,
          },
        ]
      }
      thumbImage={
        Object {
          "__packager_asset": true,
          "height": 1,
          "scale": 1,
          "uri": "file://assets/full/path/to/directory/name.type",
          "width": 1,
        }
      }
      value={0}
    />
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
          },
          Object {
            "minWidth": 20,
            "paddingLeft": 4,
          },
        ]
      }
    />
  </RNCSafeAreaProvider>
</View>
`;
